routes/api/contacts.js

```diff
@@ -1,25 +1,77 @@
-const express = require('express')
+const express = require("express");
+const Joi = require("joi");
+const {
+	listContacts,
+	getContactById,
+	removeContact,
+	addContact,
+	updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const schema = Joi.object({
+	name: Joi.string().required(),
+	email: Joi.string().email().required(),
+	phone: Joi.string().required(),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+	res.status(200).json(JSON.parse(await listContacts()));
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+	const { contactId } = req.params;
+	const id = getContactById(contactId);
+	if (id === undefined) {
+		return res.status(404).json({ message: "Not found" });
+	} else {
+		return res.status(200).json(await id);
+	}
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+	const { name, email, phone } = req.body;
+	const validation = schema.validate({ name, email, phone });
+	if (validation.error) {
+		return res.status(400).json({
+			message: `${validation.error.details[0].message}`,
+		});
+	}
+	const newContact = addContact({ name, email, phone });
+	res.status(201).json(newContact);
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.delete("/:contactId", async (req, res, next) => {
+	const { contactId } = req.params;
+	const id = removeContact(contactId);
+	if (id === false) {
+		return res.status(404).json({ message: "Not found" });
+	} else {
+		return res.status(200).json({ message: "contact deleted" });
+	}
+});
 
-module.exports = router
+router.put("/:contactId", async (req, res, next) => {
+	const { contactId } = req.params;
+	const { name, email, phone } = req.body;
+
+	if (name === "" || email === "" || phone === "") {
```

Comment:
albo here. Check by `!name | !email | !pone`. You never can be sure what type of data you get in payload and it doesn't need to be a string